### PR TITLE
systemident: set GCS perms readwrite.

### LINKED ISSUE #561: GCS: Unable to import Systemident values BUG
I can export systemident values http://i.imgur.com/kauRhgI.png but i´m somehow unable to import them successfully.

Even after manually pasting them into the fields http://i.imgur.com/5jfnHqC.png and hitting send/save, reverts them back to 0.

### PR DESCRIPTION
**Pull request by mlyle**
Sometimes it's useful to nudge things around.  Sometimes it's good to be
able to import an old tune from an exported settings set.

Fixes #561

[Review on Reviewable](https://reviewable.io/reviews/d-ronin/dronin/563)

### CODE REVIEW

**Comment by mlyle** [comment]
Unverified / untested.

**Comment by henn1001** [comment]
Tested and i can verify that this fixes the issue - thanks!

**Comment by mlyle** [comment]
@henn1001 Cool, thank you!

**Comment by mlyle** [comment]
Looks good to me too.  Ready for merge?

**Comment by mlyle** [comment]
rebased

**Comment by mlyle** [comment]
Woops! This needs to be held back because it prevents GCS from saving systemident object on autotune pane..... which it really shouldn't be trying to do even.

**Comment by mlyle** [comment]
With this additional change, things seem happy.  Turns out it was unhappy because it doesn't know how to translate some things to the uavo format (e.g. log).  But just as well anyways, this shouldn't be enabled for saving.

**Comment by mlyle** [comment]
Additional test, etc, appreciated

**Comment by mlyle** [comment]
Happy on naze32 brushed quad, through enablement to share.

**Comment by droidicus** [comment]
Tested along with #520, #411, #563, #139 on Sparky2.

Confirmed that with the sequence AT#1, export UAV, AT#2, import UAV from AT#1, power cycle, that the SystemIdent UAVO values are set to the values from AT#1.

File: ground/gcs/src/plugins/uavobjectwidgetutils/smartsavebutton.cpp [hunk 1]
```diff
@@ -158,7 +164,8 @@ void smartSaveButton::processOperation(QPushButton * button,bool save)
 
         if(upload_result==false) {
             qDebug() << "[smartsavebutton.cpp] Object upload error:" << obj->getName();
-            failedUploads.append(obj->getName());
+            if(mandatoryList.value(obj, true))
```
Comment (mluessi) [inline ground/gcs/src/plugins/uavobjectwidgetutils/smartsavebutton.cpp:167]: Why was this change necessary?

Comment (mlyle) [inline ground/gcs/src/plugins/uavobjectwidgetutils/smartsavebutton.cpp:167]: There was a long-standing bug in Tau where attempting to turn on the autotune module would return a message about being unable to save the autotune object.  This is because there was no function to convert from display format to uavo format (e.g. milliseconds to exponential tau), but we don't want to save it anyways.

I previously fixed this at TL by setting the metadata to readonly, where it catches on the other if condition and suppressed that error.  But it's sometimes nice to be able to change that value--- either to import a uavo file with a previous set of tune measurements or to be able to override inputs to the autotune stuff.  

The mandatory list was checked for permissions but wasn't checked for these other situations.  This PR changes it back to read-write, and changes other places where savings can fail to check the mandatory list.

Comment (mluessi) [inline ground/gcs/src/plugins/uavobjectwidgetutils/smartsavebutton.cpp:167]: Thanks, makes sense.

**Comment by mlyle** [comment]
Thanks for merge :)
